python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py

```diff
@@ -87,6 +87,16 @@ def df_type_check(_, value):
         success=True,
         metadata_entries=[
             EventMetadataEntry.text(str(len(value)), 'row_count', 'Number of rows in DataFrame'),
+            EventMetadataEntry.text(str(value.shape), "shape", "The shape of the DataFrame."),
+            EventMetadataEntry.md(
+                value.dtypes.to_markdown(), "dtypes", "The dtypes within the DataFrame."
+            ),
+            EventMetadataEntry.md(
+                dataframe.head().to_markdown(), "head", "A preview of the first 5 rows."
+            ),
+            EventMetadataEntry.md(
+                dataframe.tail().to_markdown(), "tail", "A preview of the last 5 rows."
```

Comment:
Out of curiosity, what is the utility behind displaying both the head and the tail?

Reply:
@asingh16 not sure about your experience, but my experience is that early data often looks very different than recent data. At least in our use cases, it's often nice to be able to have a preview of both.